### PR TITLE
add access point support

### PR DESCRIPTION
**Pull request by jurgenweber**
## what
* added aws efs access point support 

## why
*efs supports access points within a volume
* from a performance PoV using burst, you are better off to have one large EFS vol.

## references
* https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/efs_access_point

### CODE REVIEW

File: outputs.tf [hunk 1]
```diff
@@ -1,3 +1,13 @@
+output "access_point_arn" {
```
Comment (aknysh) [inline outputs.tf:1]:
```suggestion
output "access_point_arns" {
```

File: outputs.tf [hunk 2]
```diff
@@ -1,3 +1,13 @@
...
+  value       = { for arn in sort(keys(var.access_points)) : arn => aws_efs_access_point.default[arn].arn }
+  description = "EFS AP ARNs"
+}
+
+output "access_point_id" {
```
Comment (aknysh) [inline outputs.tf:6]:
```suggestion
output "access_point_ids" {
```

File: variables.tf [hunk 3]
```diff
@@ -9,6 +9,12 @@ variable "allowed_cidr_blocks" {
   description = "The CIDR blocks from which to allow `ingress` traffic to the EFS"
 }
 
+variable "access_points" {
+  type        = map(map(map(any)))
+  default     = {}
+  description = "A list of the access points you would like in your EFS volume"
```
Comment (aknysh) [inline variables.tf:15]: maybe update the description since it says `a list` but the data type is a map

**Review by aknysh — COMMENTED** [review]
thanks @jurgenweber 
please see comments

**Review by aknysh — COMMENTED** [review]
thanks @jurgenweber 
LGTM, but please see the comments to update the output names

**Comment by jurgenweber** [comment]
Cool! Did you note this? https://github.com/cloudposse/terraform-aws-efs/pull/58/files#diff-7a370d8342e7203b805911c92454f0f4L36-R63

I was not sure about this ,but considering the TF 13.3 rls notes (https://github.com/hashicorp/terraform/releases/tag/v0.13.3) I think maybe I have disabled this as a work around to the bug.

I am going to test with and without this lifecycle stanza deleting in 13.3 and see if I get the same cycle error shortly.

**Comment by jurgenweber** [comment]
Yea, so I have confirmed with TF 0.13.3 that error I was getting is now gone. So I have put that lifecycle block back.

**Comment by jurgenweber** [comment]
@aknysh ready when you are

**Review by aknysh — APPROVED** [review]

**Pull request merged**